lib/block-supports/spacing.php

```diff
@@ -45,33 +45,46 @@ function gutenberg_apply_spacing_support( $block_type, $block_attributes ) {
 
 	$has_padding_support = gutenberg_block_has_support( $block_type, array( 'spacing', 'padding' ), false );
 	$has_margin_support  = gutenberg_block_has_support( $block_type, array( 'spacing', 'margin' ), false );
+	$block_styles        = isset( $block_attributes['style'] ) ? $block_attributes['style'] : null;
+	$classes             = array();
 	$styles              = array();
+	$style_engine        = WP_Style_Engine_Gutenberg::get_instance();
 
 	if ( $has_padding_support ) {
-		$padding_value = _wp_array_get( $block_attributes, array( 'style', 'spacing', 'padding' ), null );
-
-		if ( is_array( $padding_value ) ) {
-			foreach ( $padding_value as $key => $value ) {
-				$styles[] = sprintf( 'padding-%s: %s;', $key, $value );
-			}
-		} elseif ( null !== $padding_value ) {
-			$styles[] = sprintf( 'padding: %s;', $padding_value );
+		$padding_styles = $style_engine->generate(
+			$block_styles,
+			array( 'inline' => true )
+		);
+
+		if ( $padding_styles ) {
+			$styles[] = $padding_styles;
 		}
 	}
 
 	if ( $has_margin_support ) {
-		$margin_value = _wp_array_get( $block_attributes, array( 'style', 'spacing', 'margin' ), null );
-
-		if ( is_array( $margin_value ) ) {
-			foreach ( $margin_value as $key => $value ) {
-				$styles[] = sprintf( 'margin-%s: %s;', $key, $value );
-			}
-		} elseif ( null !== $margin_value ) {
-			$styles[] = sprintf( 'margin: %s;', $margin_value );
+		// As with padding above.
+		$margin_styles = $style_engine->generate(
+			$block_styles,
+			array( 'inline' => true )
+		);
```

Comment:
It looks like the style engine in this PR is designed to only be called once per block, so it might become a problem when we roll out to other block supports, too.

I suppose we could either `unset` attributes that haven't opted in for support, or defer the support check to the style engine? I think I'd lean toward keeping the support checks within the block support, and just call `generate` once 🤔

Reply:
I committed a bunch of code without test thinking I was still in draft mode. Apologies and thanks for reporting. Getting the rest of the working code up soon...